src/Form/DataTransformer/ModelToIdPropertyTransformer.php

```diff
@@ -130,7 +131,12 @@ public function transform($entityOrCollection)
         }
 
         foreach ($collection as $entity) {
-            $id = current($this->modelManager->getIdentifierValues($entity));
+            $ids = $this->modelManager->getIdentifierValues($entity);
+            if (count($ids) > 1) {
+                $id = implode(AdapterInterface::ID_SEPARATOR, $ids);
```

Comment:
if It is always an array, this should be also an array right? because now it will be an string